pkgs/development/tools/language-servers/vhdl-ls/default.nix

```
@@ -0,0 +1,43 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "vhdl-ls";
+  version = "0.64.0";
+
+  src = fetchFromGitHub {
+    owner = "VHDL-LS";
+    repo = "rust_hdl";
+    rev = "v${version}";
+    hash = "sha256-j5WRJJBUPKW3W+kY5hdqdZxxGkIAoEcW+A2pp23MX6Q=";
+  };
+
+  # No Cargo.lock upstream, see:
+  # https://github.com/VHDL-LS/rust_hdl/issues/166
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+  };
+  postPatch = ''
+    ln -s ${./Cargo.lock} Cargo.lock
+  ''
+  # Also make it look up vhdl_libraries in an expected location
+  + ''
+    substituteInPlace vhdl_lang/src/config.rs \
+      --replace /usr/lib $out/lib
+  '';
+
+  postInstall = ''
+    mkdir -p $out/lib/rust_hdl
+    cp -r vhdl_libraries $out/lib/rust_hdl
+  '';
+
+  meta = with lib; {
+    description = "a fast VHDL language server";
```

```suggestion
    description = "A fast VHDL language server and analysis library written in Rust";
```
Only commenting because first letter should be uppercase. The `written in Rust`, I don't care. Took from upstream.